pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -1019,7 +1013,7 @@ policy:
 	}
 	ee = resiliency.NewErrorSet("failed to delete XFRM states", len(xfrmStateList))
 	for _, s := range xfrmStateList {
-		if isXfrmStateCilium(s) && (reqID == 0 || s.Reqid == reqID) {
```

Comment:
I expected you to just do inlining of `DeleteXFRM` into the call sites and rename `DeleteXFRMWithReqID` to `DeleteXFRM`. But that's not what you're actually doing here, right?

`DeleteXFRM` used to delete everything thanks to this condition. Now `DeleteXFRM(log, DefaultReqID)` is going to delete all rules with ReqID 1. But not all XFRM rules have ReqID 1, so the behavior changed here? Or am I missing something?

Reply:
@pchaigno updated with latest force push